tests/test_environments/test_harbor_timeouts.py

```
@@ -92,3 +92,44 @@ async def test_verify_uses_1800_when_task_declares_it(tmp_path):
+    cmd = sb.test_cmds[0]
+    # JAVA_HOME must be guarded so an unset value cannot become a bare /bin
+    assert ":$JAVA_HOME/bin:$PATH" not in cmd  # the old unguarded form
+    assert "${JAVA_HOME:+:$JAVA_HOME/bin}" in cmd
+    # image PATH ($PATH) must precede the prepended toolchain dirs so the
+    # image's python/node/etc. win
+    export = cmd.split("&&")[0]
+    assert export.index("$PATH") < export.index("/root/.local/bin")
+    # the assembled command must be valid shell (balanced quotes)
+    import subprocess as _sp
+
+    assert _sp.run(["bash", "-n", "-c", cmd], capture_output=True).returncode == 0
```

Ruff (0.16.0)

[warning] Async functions should not run processes with blocking methods

(ASYNC221)